exercises/practice/bob/.approaches/answer-array/content.md

```diff
@@ -0,0 +1,41 @@
+# Answer array
+
+```csharp
+    // uses System.Linq;
+    private static readonly string[] answers = {"Whatever.", "Sure.", "Whoa, chill out!", "Calm down, I know what I'm doing!"};
+    
+    public static string Response(string statement)
+    {
+        var input = statement.TrimEnd();
+        if (input == "") return "Fine. Be that way!";
```

Comment:
I think it makes sense to use official conventions by default, but I really like the idea of having a "shortening" section to explore things that are totally legal in the syntax of the language, but perhaps less common, or something where there are camps and people tend to fall on one side of the other.

Reply:
>  I chafe a bit at "official style guidelines". I'm more of a "I'll take your advice under advisement and do what seems right to me" kinda person, because I find it less tedious than looking up guidelines for everything, but I suppose I can bend to them for the sake of consistency.

I can see where you are coming from and we should not be dogmatic about it, but there _is_ value in using a syntax that will be familiar to most. In this case, I'd argue that the `if` and the `return` on the same line is not commonly used in C# and thus easy to miss for students.

> As for curly braces, I understand why languages such as Go and Rust require them to make scoping explicit, but for single-line expressions they're just more syntactic noise.

I agree.

>  It's the same reason I prefer member-bodied expressions. Just say no to cruft! 

Absolutely! I always write member-bodied expressions, I'm just trying to consider the other C# students :)

Would you object to using:

```csharp
if (input == "")
    return "Fine. Be that way!";
```
?

Reply:
> I'm more of a "I'll take your advice under advisement and do what seems right to me" kinda person, because I find it less tedious than looking up guidelines for everything

So just to be clear: I don't want us to enforce a style or anything, but I would like us to use a somewhat consistent style across our approaches and it should be a style that is easily accessible to students. As most C# I read seems to be fairly similar to the official style guide, I would like us to at least try and match that style. We probably need to figure this out feature by feature (and maybe document this?).

I also quite like like a `Shortening` section as it can be a fun way to try and make things as concise as possible (which could reduce readability).